src/main/java/com/epam/dataservice/HomeWork3.java

```diff
@@ -0,0 +1,154 @@
+package com.epam.dataservice;
+
+import com.epam.data.RoadAccident;
+import com.epam.data.RoadAccidentBuilder;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by bill on 16-5-1.
+ */
+public class HomeWork3 {
+    private final int MAX_READ_THREADS = 2;
+    private final int MAX_PROC_THREADS = 4;
+    public static final int MAX_ENRICH_THREADS = 100;
+    private final int MAX_WRITE_THREADS = 2;
+    public static final int MAX_BATCH_SIZE = 40000;
+    public static final String outputFileNames[] = {"NighttimeAccidents.csv", "DaytimeAccidents.csv"};
+
+    private LinkedList<String> fileList;
+    private BlockingQueue<List<RoadAccident>> readQueue;
+    private ExecutorService readExecutor;
+    private List<Future<Integer>> readTaskList = new ArrayList<>();
+
+    private ExecutorService procExecutor;
+
+    private List<BlockingQueue<RoadAccident>> writeQueues = new ArrayList<>(MAX_WRITE_THREADS);
+    private ExecutorService writeExecutor;
+
+    public HomeWork3(LinkedList<String> fileList) {
+        this.fileList = fileList;
+    }
+
+    public void startTask() {
```

Comment:
Sorry - again about clean code :smile: 
This method is a bit hard to read when you read it for very first itme. It would be much easier to understand if you split it on several methods like runFileLoaders, runDataProcessors, runBatchWriters...